tests-ci/tests/test_storage.js

```diff
@@ -0,0 +1,352 @@
+      'child app storage differs');
+  }).catch(() => imports.assert(false, 'child app test failed'));
+};
+
+/**
+ * Apps can't exceed quota.
+ * Create child scene with quota set via 'permissions'. Set items (by child) exceeding quota fails.
+ * @returns {Promise<string>}
+ */
+module.exports.tests.test10_childAppStoragePermissions = () => {
+  const storage = imports.scene.storage;
+
+  storage.clear();
+  storage.setItem('key1', 'abc');
+  storage.setItem('key2', 'def');
+
+  const new_scene = imports.scene.create({
+    t: 'scene',
+    parent: imports.scene.root,
+    w: imports.scene.w, h: imports.scene.h,
+    url: child_url,
+    permissions: {
+      'url' : {
+        'allow' : [ '*' ]
+      },
+      "storage": {
+        "allow": 10
+      }
+    }
+  });
+
+  return new_scene.ready.then(s => {
+    const child = s.api.getStorage();
+
+    try {
+      child.setItem('key3', 'ghi');
+      return imports.assert(false, 'quota test');
+    } catch (ignored) {
+    }
+
+    const value1 = child.getItem('key1');
+    const value2 = child.getItem('key2');
+    const value3 = child.getItem('key3');
+
+    return imports.assert(
+      value1 === 'abc' && value2 === 'def' && value3 === '',
+      'child app storage differs');
+  }).catch(() => imports.assert(false, 'child app quota test failed'));
+};
+
+/**
+ * Random apps have 0 quota.
+ * Create child scene (random origin). Set items (by child) fails. 'getItems' (by child) gives an empty array.
+ * @returns {Promise<string>}
+ */
+module.exports.tests.test11_arbitraryAppQuota = () => {
+  const new_scene = imports.scene.create({
+    t: 'scene',
+    parent: imports.scene.root,
+    w: imports.scene.w, h: imports.scene.h,
+    url: tiny_url
+  });
+
+  return new_scene.ready.then(s => {
+    const child = s.api.getStorage();
+
+    child.clear();
+
+    try {
+      child.setItem('key', 'value');
+      return imports.assert(false, 'quota test');
+    } catch (ignored) {
+    }
+
+    const value = child.getItems();
+
+    return imports.assert(value.length === 0, 'arbitrary app storage is not empty');
+  }).catch(() => imports.assert(false, 'arbitrary app test failed'));
+};
+
+/**
+ * Quota test.
+ * Set items (by child app) exceeding quota fails, not exceeding - succeeds. Remove frees space.
+ * @returns {Promise<string>}
+ */
+module.exports.tests.test12_quota = () => {
+  const new_scene = imports.scene.create({
+    t: 'scene',
+    parent: imports.scene.root,
+    w: imports.scene.w, h: imports.scene.h,
+    url: child_url,
+    permissions: {
+      'url' : {
+        'allow' : [ '*' ]
+      },
+      "storage": {
+        "allow": 20
+      }
+    }
+  });
+
+  return new_scene.ready.then(s => {
+    const child = s.api.getStorage();
+
+    child.clear();
+    child.setItem('1234', '5678');
+    child.setItem('9012', '3456');
+    // len 16
+    try {
+      child.setItem('7890', '1234');
+      return imports.assert(false, 'quota test');
```

Comment:
Please make each of these failure messages unique - even with just a number - so that someone debugging will be able to tell exactly which part of this test failed it they get a failure.